### PR TITLE
Left column selections foundation + service updates

### PR DESCRIPTION
**Pull request by FarajiA**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-392

 ## Description:
This PR is the ground work for user based selections of systems, measurements, and devices. The page will load a default set of graphs and then behave based on selections.

All updates to the URL are made through the _Container Component_ which then passes on this data to the _Selection Component_ via attributes of the component.

The _Graph Component_ will then load graphs based on the selections being made and/or the URL being loaded.

The _MetricsService_ houses the observables for the selections being made and then reports them to the graph components. 

Visualize-Page

 ## Testing:
`npm run test`

 ## Screenshots:
visualizeGraphs

### CODE REVIEW

**Comment by itzg** [comment]
I just wanted to make a quick note based on the screenshot. The need to pick the monitoring system makes total sense given how the metrics are currently organized and queried; however, I think it makes the usability for our end users a little awkward since they need to know what system provides what metric. Furthermore, as we migrate monitoring systems, such as MaaS to Salus, the choice monitoring system is going to be a changing answer.

**Comment by FarajiA** [comment]
Ah I follow. It may not be apparent to an end user where to initially look for a particular metric or recent alert. Which is why now that I think about it, Intelligence has that section that lets you look at your graphs based on check.

graphsByCheck

Okay, I can think about ways in which to query for something on the basis of category. I think this will help users navigate through what they're seeing.

**Comment by itzg** [comment]
As food for thought, it might be worth poking around in Stackdriver's UI to see how they address the potential of lots of resource types, custom metrics, and metrics overall. Basically, they seem to present a long list of choices with robust text filtering along with filtering by resources, once selected:

image

**Comment by FarajiA** [comment]
Okay after some updating here's the latest screenshot.
comboBox

The goal is try use a lot more Comboboxes for looking up a resource or in the future a measurement.

File: src/app/_features/visualize/components/selections/selections.component.ts [hunk 1]
```diff
@@ -8,14 +11,114 @@ import { MetricsService } from '../../../../_services/metrics/metrics.service';
...
+   * @param {IMeasurement} measurement
+   */
+  selectMeasurement(measurement: IMeasurement): void {
+    this.measurement = measurement.name;
+    // based on the selected metric we want to get the fields associated and the devices
+    this.metricService.getMetricFields(this.measurement).pipe(
+      flatMap(fields => {
+        this.selectedFields = fields;
+        return this.metricService.getDevices(fields[0].fieldKey, this.measurement,
+          '6h', 'now()');
```
Comment (itzg) [inline src/app/_features/visualize/components/selections/selections.component.ts]: Would be good to declare the query time range in a "config object" or even just constants at the top of this class.

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts [hunk 2]
```diff
@@ -17,25 +17,85 @@ export class VisualizePage implements OnInit {
...
+    this.subManager.unsubscribe();
   }
 
+  /**
+   * @description this function updates the URL query params
+   * @return {void}
+   */
+  private updateQueryParams() {
+    const params = {
+      ...(this.system  && { system: this.system }),
```
Comment (itzg) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts]: Oh, I was actually wondering recently how to do an optional spread without resorting to if-else blocks.

File: src/app/_features/visualize/components/selections/selections.component.ts (hunk 1 above)
Comment (FarajiA) [reply src/app/_features/visualize/components/selections/selections.component.ts]: Yup, updated this.

**Pull request merged**